extension/mozcentral/browser/extensions/shumway/content/ShumwayUtils.jsm

```diff
@@ -56,6 +58,19 @@ let ShumwayUtils = {
     Services.prefs.addObserver(PREF_DISABLED, this, false);
   },
 
+  migratePreferences: function migratePreferences() {
+    // At one point we had shumway.disabled set to true by default,
+    // and we are trying to replace it with shumway.swf.whitelist:
+    // checking if the user already changed it before to reset
+    // the whitelist to '*'.
+    if (Services.prefs.prefHasUserValue(PREF_DISABLED) &&
+        !Services.prefs.prefHasUserValue(PREF_WHITELIST) &&
+        !getBoolPref(PREF_DISABLED, false)) {
+      // The user is already using Shumway -- enabling all web sites.
+      Services.prefs.setCharPref(PREF_WHITELIST, '*');
```

Comment:
I don't think we should do this. It'd mean that we can't ever get these users to the whitelist we want them to have. Instead, we should do something like
```js
if (this.enabled)
  this._ensureRegistered(getWhitelist());
```
Where `migratePreferences` replaced by a `getWhitelist` that just returns `'*'` for the users that have the `disabled` flag toggled, and the real whitelist from the config for everyone else.